types/mongoose/index.d.ts

```diff
@@ -75,6 +75,12 @@ declare module "mongoose" {
   import stream = require('stream');
   import mongoose = require('mongoose');
 
+  // We can use TypeScript Omit once minimum required TypeScript Version is above 3.5
+  type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
+
+  /* Helper type to extract a definition type from a Document type */
+  type DocumentDefinition<T> = Omit<T, keyof Document> & { _id: mongodb.ObjectId };
```

Comment:
@avaly Wouldn't it be wiser to use `T['_id']` ? This release broke our build because we're using `_id: string` in our models.

Reply:
Or even just the omit part, I don't think the union is necessary here.

Reply:
Sounds like a decent thing to do. I'll PR this change shortly.

Reply:
My last suggestion likely won't work because MongooseDocument defines a `_id: any`, so rather pick the 1st one.

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43039

Reply:
Thanks a lot!